crates/proof/src/preloaded_eigenda_provider.rs

```diff
@@ -14,7 +18,7 @@ use tracing::info;
 #[derive(Clone, Debug, Default)]
 pub struct PreloadedEigenDABlobProvider {
```

Comment:
can you add documentation here? You've documented the invariants it respects, but not how its used. I'm guessing this is the guy that's preloaded from the first derivation pipeline run on the native client, and passed to the zkvm? Would be good to document this